syntax = "proto3";

message Node {
  message Location {
    float x = 1;
    float y = 2;
  }

  int64 id = 1;
  Location location = 2;
  float importance = 3;
}

message WriteUndirectedEdgeRequest {
  Node node_a = 1;
  Node node_b = 2;
  float distance = 3;
}

message DeleteUndirectedEdgeRequest {
  int64 node_a_id = 1;
  int64 node_b_id = 2;
}

message MutationRequest {
  oneof request {
    WriteUndirectedEdgeRequest write_edge = 1;
    DeleteUndirectedEdgeRequest delete_edge = 2;
  }
}

message MutationRequests { repeated MutationRequest requests = 1; }